### PR TITLE
Theming v2 channel search

### LINKED ISSUE #1580: Apply theme-v2 to search component
**Motivation**
A clear and concise description of what the problem/opportunity is. Ex. I'm always frustrated when [...]

**Proposed solution**

**Acceptance Criteria**
- [ ] Layout
- [ ] Theme
- [ ] RTL support
- [ ] Component layer variable docs

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Add channel search functionality according to theme v2 designs.
Closes #1580.
Depends on https://github.com/GetStream/stream-chat-css/pull/157.

### 🛠 Implementation details

1. **Extract the channel search logic into a hook.** 
The reason for this was: The v2 designs show the search results to be rendered instead of the list of loaded channels. Original implementation did not allow to display either search results or list of loaded channels, because the channel search query results were not known in the `ChannelList` component - only in the `ChannelSearch` component.

2. **Add new `SearchBar` component**
Implemented default component as well as added a new `ChannelList` prop `SearchBar`.

3. **Backwards compatibility**
Only new configuration parameters for the `ChannelList` prop `additionalChannelSearchProps` were added

4. **Channel search types refactor** 
Original `ChannelSearchProps` type was decomposed into props types that make up the `ChannelSearch` component (props for input + props for search results + to keep backwards compatibility props for channel search controller - hook)

5. **Use `ChannelSearch` for old theme, `SearchBar` for theme v2**
Besides that the `ChannelSearch` component is used also for scenarios, when search results are to be displayed in context menu floating above the list of loaded channels in the `ChannelList`. This is because the v2 design does not count with the floating alternative.

### Out of scope
Sidebar / app context menu - only boilerplate for plugging in custom `AppMenu` component created

image

### Still todo

- [x] Add translations for search results header
- [x] Sidebar / app context menu button active state

### 🎨 UI Changes

|What | Figma | v2 | v1|
|-------|-----------|----|----|
|Search inactive | image | image | image|
|Focus the search input | N/A | image | image |
|Search query in progress | N/A | image | image |
|No search results | image | image | image |
|Search error | image | N/A | N/A |
|Show x > 0 search results (channels + users) | image | image | image|
|Select result that is not the active channel && clearSearchOnClickOutside === false | image | image | image |
|Click the X (clear) button (without selecting a result) | N/A | image | N/A |
|Click the X (clear) button (after selecting a non-active channel) - adds new channel to the top - not optimal, but follows the already existing logic | N/A | image | N/A |
|Click the "back / exit" button | N/A | image | N/A |
|Non-channel search results (users/contacts) | N/A | image | image |

### CODE REVIEW

File: src/components/ChannelList/ChannelList.tsx [hunk 1]
```diff
@@ -183,8 +216,23 @@ const UnMemoizedChannelList = <
     navOpen = false,
     setActiveChannel,
     theme,
+    themeVersion,
     useImageFlagEmojisOnWindows,
   } = useChatContext<StreamChatGenerics>('ChannelList');
+  console.log('active', channel);
```
Comment (arnautov-anton) [inline src/components/ChannelList/ChannelList.tsx:222]:
```suggestion
```

Comment (MartinCupela) [reply src/components/ChannelList/ChannelList.tsx]: thank you @arnautov-anton , I have removed the console.log call

**Pull request merged**